src/Database/Schema/ForeignKey.php

```diff
@@ -185,4 +196,50 @@ protected function normalizeAction(string $action): string
         }
         throw new InvalidArgumentException('Unknown action passed: ' . $action);
     }
+
+    /**
+     * Sets deferrable mode for the foreign key.
+     *
+     * @param string $deferrable Constraint
+     * @return $this
+     */
+    public function setDeferrable(string $deferrable)
+    {
+        $this->deferrable = $this->normalizeDeferrable($deferrable);
```

Comment:
What would set an incorrect value here that's valid when read back?

Reply:
Nothing, but a developer could try `$key->setDeferrable('blah')` which is what I was trying to prevent here.